app/actions/remote/entry/common.ts

```diff
@@ -188,16 +191,45 @@ const fetchAppEntryData = async (serverUrl: string, sinceArg: number, initialTea
         }
     }
 
-    // Fetch in parallel teams / team membership / channels for current team / user preferences / user
-    const promises: [Promise<MyTeamsRequest>, Promise<MyChannelsRequest | undefined>, Promise<MyUserRequest>] = [
+    // Fetch in parallel teams / team membership / user preferences / user
+    const promises: [Promise<MyTeamsRequest>, Promise<MyUserRequest>] = [
         fetchMyTeams(serverUrl, fetchOnly),
-        initialTeamId ? fetchMyChannelsForTeam(serverUrl, initialTeamId, includeDeletedChannels, since, fetchOnly, false, isCRTEnabled) : Promise.resolve(undefined),
         fetchMe(serverUrl, fetchOnly),
     ];
 
     const resolution = await Promise.all(promises);
-    const [teamData, , meData] = resolution;
-    let [, chData] = resolution;
+    const [teamData, meData] = resolution;
+    let chData;
+
+    let initialTeamId = onLoadTeamId;
+    let initialChannelId = channelId;
+    let gmConverted = false;
+
+    if (channelId) {
```

Comment:
@larkox @enahum I've done some changes here so we don't call `fetchMyChannelsForTeam` only when needed, instead of always calling it for the incorrect team previously at line 194.

Reply:
I have mixed feelings with this change.

How often a GM will be converted? Not often. So we will be "delaying" and doing one extra call on the happy path to avoid calling twice `fetchMyChannelsForTeam` once in a blue moon. So... 0/5 on this change.